netlogo-gui/src/main/window/AutoSuggest.scala

```diff
@@ -0,0 +1,218 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.window
+
+import org.nlogo.core.DefaultTokenMapper
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.collection.immutable.TreeSet
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/**
+  * Builds the trie from commands and reporters and provides fuctions to get suggestions.
+  * Special care should be taken as everything is inserted and retrieved in lower case.
+  */
+class AutoSuggest {
+  val commandNames = DefaultTokenMapper.allCommandNames
+  val reporterNames = DefaultTokenMapper.allReporterNames
+
+  val trie = new TrieNode()
+  for(commandName <- commandNames) {
+    trie.append(commandName.toLowerCase)
+  }
+  for(reporterName <- reporterNames) {
+    trie.append(reporterName.toLowerCase)
+  }
+
+  def editDistance(s1: String, s2: String): Int = {
+    val memo = scala.collection.mutable.Map[(List[Char],List[Char]),Int]()
+    def min(a:Int, b:Int, c:Int) = Math.min( Math.min( a, b ), c)
+    def sd(s1: List[Char], s2: List[Char]): Int = {
```

Comment:
What does `sd` mean? I would prefer a more descriptive name unless this is a domain-specific term (if it is just let me know).